feat(live): add a new resource to generate a sign URL (#6064)
huaweicloud/terraform-provider-huaweicloud: docs/resources/live_url_authentication.md

---
subcategory: "Live"
layout: "huaweicloud"
page_title: "HuaweiCloud: huaweicloud_live_url_authentication"
description: |-
  Manages an URL authentication resource within HuaweiCloud.
---

# huaweicloud_live_url_authentication

Manages an URL authentication resource within HuaweiCloud.

> -> The current resource is a one-time resource, and destroying this resource will not change the current status.

Reviewer: Does repeatedly applying this resource have any impact?

  type        = var.type
  stream_name = var.stream_name
  app_name    = var.app_name
}
```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
  If omitted, the provider-level region will be used.
  Changing this parameter will create a new resource.

* `domain_name` - (Required, String, ForceNew) Specifies the domain name to which the URL validation belongs.
  Including the ingest domain name and streaming domain name.
  Changing this parameter will create a new resource.

* `type` - (Required, String, ForceNew) Specifies the type of the domain name.
  The valid values are as follow:
  + **push**: Indicates ingest domain name.
  + **pull**: Indicates streaming domain name.

  Changing this parameter will create a new resource.

* `stream_name` - (Required, String, ForceNew) Specifies the stream name.
  Changing this parameter will create a new resource.

* `app_name` - (Required, String, ForceNew) Specifies the application name.
  Changing this parameter will create a new resource.

* `check_level` - (Optional, Int, ForceNew) Specifies the check level.

>   This parameter is valid and mandatory when the value of the `auth_type` in the URL validation is set to **c_aes**.

Reviewer 2: ```suggestion
  This parameter is valid and mandatory only when the signing method is **c_aes** in URL validation.
```
